src/flint/types/fq_default.pyx

```diff
@@ -0,0 +1,886 @@
+        """
+        Return the unit element
+
+            >>> gf = fq_default_ctx(5)
+            >>> gf.one()
+            1
+        """
+        cdef fq_default res
+        res = self.new_ctype_fq_default()
+        res.ctx = self
+        fq_default_one(res.val, self.val)
+        return res
+
+    def gen(self):
+        """
+        Return the one element
+
+            >>> gf = fq_default_ctx(5, 2, var="w")
+            >>> gf.gen()
+            w
+        """
+        cdef fq_default res
+        res = self.new_ctype_fq_default()
+        res.ctx = self
+        fq_default_gen(res.val, self.val)
+        return res
+
+    def random_element(self, not_zero=False):
+        r"""
+        Return a random element of the finite field
+
+            >>> gf = fq_default_ctx(163, 3)
+            >>> a = gf.random_element()
+            >>> type(a) is fq_default
+            True
+            >>> a = gf.random_element(not_zero=True)
+            >>> not a.is_zero()
+            True
+        """
+        cdef fq_default res
+        res = self.new_ctype_fq_default()
+        res.ctx = self
+        if not_zero:
+            fq_default_rand_not_zero(res.val, global_random_state, self.val)
+        else:
+            fq_default_rand(res.val, global_random_state, self.val)
+        return res
+
+    cdef new_ctype_fq_default(self):
+        return fq_default.__new__(fq_default, None, self)
+
+    cdef set_list_as_fq_default(self, fq_default_t fq_ele, obj):
+        cdef fmpz_poly poly
+        poly = fmpz_poly.__new__(fmpz_poly)
+        fmpz_poly_set_list(poly.val, obj)
+
+        # Now set the value from the fmpz_poly
+        fq_default_set_fmpz_poly(fq_ele, poly.val, self.val)
+
+        return 0
+
+    cdef set_any_scalar_as_fq_default(self, fq_default_t fq_ele, obj):
+        if typecheck(obj, int):
+            # For small integers we can convert directly
+            if obj < 0 and obj.bit_length() < 31:
+                fq_default_set_si(fq_ele, <slong>obj, self.val)
+            elif obj.bit_length() < 32:
+                fq_default_set_ui(fq_ele, <ulong>obj, self.val)
+            # For larger integers we first convert to fmpz
+            else:
+                obj_fmpz = any_as_fmpz(obj)
+                fq_default_set_fmpz(fq_ele, (<fmpz>obj_fmpz).val, self.val)
+            return 0
+
+        # For fmpz we can also convert directly
+        if typecheck(obj, fmpz):
+            fq_default_set_fmpz(fq_ele, (<fmpz>obj).val, self.val)
+            return 0
+
+        # For nmod we can convert by taking it as an int
+        # Ignores the modulus of nmod
+        if typecheck(obj, nmod):
+            fq_default_set_ui(fq_ele, <ulong>(<nmod>obj).val, self.val)
+            return 0
+
+        # For fmpz_mod we can also convert directly
+        # Assumes that the modulus of the ring matches the context
+        if typecheck(obj, fmpz_mod):
+            fq_default_set_fmpz(fq_ele, (<fmpz_mod>obj).val, self.val)
+            return 0
```

Comment:
If this function is being used by `__add__` etc then the modulus needs to be checked:
```python
In [70]: F = fq_default_ctx(3)

In [71]: F(2) + nmod(2, 7)
Out[71]: 1
```

Reply:
oh i thought i handled this... i must have forgotten to do it.